PickMeUp/PickMeUp/viewControllers/DM Features/PMConversationViewController.m

```diff
@@ -78,35 +110,47 @@ -(void)_runGetQuery {
         self.noDMs = NO;
         PFQuery *getQuery = [PFQuery queryWithClassName:kDirectMessageClassName];
         [getQuery whereKey:kConvoIdKey equalTo:self.convo.objectId];
+        getQuery.limit = 30;
+        [getQuery orderByDescending:kCreatedAtKey];
+        if (self.arrayOfDMs != nil) {
+            getQuery.skip = self.arrayOfDMs.count;
+        }
         [getQuery findObjectsInBackgroundWithBlock:^(NSArray *DMs, NSError *error) {
             if (DMs != nil) {
```

Comment:
FYI can use NSAssert here to make sure this block is called on the main thread
`NSAssert([[NSThread currentThread] isMainThread], @"must be called on the main thread")`

Might be beneficial, because I find this parse ambiguous a bit, since some of its callback are called on the main thread and some are not. That way you can catch bugs during development.